api/cloudcontroller/ccv2/service_plan_visibility_test.go

```diff
@@ -131,7 +131,145 @@ var _ = Describe("Service Plan Visibility", func() {
+		})
+
+		When("the cc returns no error", func() {
+			BeforeEach(func() {
+				request := map[string]string{
+					"service_plan_guid": "plan-guid-1",
+					"organization_guid": "org-guid-1",
+				}
+
+				response := `{
+				"metadata": {
+					"guid": "plan-visibility-1",
+					"url": "/v2/service_plan_visibilities/f740b01a-4afe-4435-aedd-0a8308a7e7d6",
+					"created_at": "2016-06-08T16:41:31Z",
+					"updated_at": "2016-06-08T16:41:26Z"
+				},
+				"entity": {
+					"service_plan_guid": "plan-guid-1",
+					"organization_guid": "org-guid-1",
+					"service_plan_url": "/v2/service_plans/ab5780a9-ac8e-4412-9496-4512e865011a",
+					"organization_url": "/v2/organizations/55d0ff39-dac9-431f-ba6d-83f37381f1c3"
+				}
+			}`
+
+				server.AppendHandlers(
+					CombineHandlers(
+						VerifyRequest(http.MethodPost, "/v2/service_plan_visibilities"),
+						VerifyJSONRepresenting(request),
+						RespondWith(http.StatusCreated, response, http.Header{"X-Cf-Warnings": {"warning-1, warning-2"}}),
+					),
+				)
+			})
+
+			It("returns the created service plan visibility", func() {
+				Expect(executeErr).NotTo(HaveOccurred())
+				Expect(visibility.GUID).To(Equal("plan-visibility-1"))
+				Expect(visibility.ServicePlanGUID).To(Equal("plan-guid-1"))
+				Expect(visibility.OrganizationGUID).To(Equal("org-guid-1"))
+				Expect(warnings).To(ConsistOf("warning-1", "warning-2"))
+			})
+		})
+
+		When("the cc returns an error", func() {
+			BeforeEach(func() {
+				response := `{
+					"code": 10001,
+					"description": "Some Error",
+					"error_code": "CF-SomeError"
+				}`
+				server.AppendHandlers(
+					CombineHandlers(
+						VerifyRequest(http.MethodPost, "/v2/service_plan_visibilities"),
+						RespondWith(http.StatusTeapot, response, http.Header{"X-Cf-Warnings": {"warning-1, warning-2"}}),
+					),
+				)
+			})
+
+			It("returns the error and warnings", func() {
+				Expect(executeErr).To(MatchError(ccerror.V2UnexpectedResponseError{
+					ResponseCode: http.StatusTeapot,
+					V2ErrorResponse: ccerror.V2ErrorResponse{
+						Code:        10001,
+						Description: "Some Error",
+						ErrorCode:   "CF-SomeError",
+					},
+				}))
+				Expect(warnings).To(ConsistOf("warning-1", "warning-2"))
+			})
+		})
+	})
+
+	Describe("DeleteServicePlanVisibility", func() {
+		var (
+			warnings   Warnings
+			executeErr error
+		)
+
+		JustBeforeEach(func() {
+			warnings, executeErr = client.DeleteServicePlanVisibility("service-plan-visibility-guid")
+		})
+
+		When("the cc returns no error", func() {
+			BeforeEach(func() {
+				response := `{}`
+
+				server.AppendHandlers(
+					CombineHandlers(
+						VerifyRequest(http.MethodDelete, "/v2/service_plan_visibilities/service-plan-visibility-guid"),
+						RespondWith(http.StatusNoContent, response, http.Header{"X-Cf-Warnings": {"warning-1, warning-2"}}),
+					),
+				)
+			})
+
+			It("returns the created service plan visibility", func() {
+				Expect(server.ReceivedRequests()).Should(HaveLen(2))
```

Comment:
Why 2?

Reply:
The client makes `GET` to `/v2/info` first and then the actual `DELETE`.

We already make the verification that we want as part of the server handler, so I guess this expectation is a bit redundant.